rollup/pipeline/pipeline.go

```diff
@@ -309,25 +332,70 @@ type Result struct {
 	FinalBlock *BlockCandidate
 }
 
+func (p *Pipeline) encodeStage(traces <-chan *BlockCandidate) <-chan *BlockCandidate {
+	downstreamCh := make(chan *BlockCandidate, p.downstreamChCapacity())
+	p.wg.Add(1)
+	buffer := new(bytes.Buffer)
```

Comment:
Why assigning it outside the scope of the goroutine?

Reply:
no particular reason, moving it to go routine scope would be more correct.

Reply:
done in https://github.com/scroll-tech/go-ethereum/pull/814/commits/44d70f89f26b77f1a2842c8dfb11217a2255aaee